packages/astro/src/compat/index.ts

```diff
@@ -0,0 +1,34 @@
+import type { LogOptions } from '../core/logger/core';
+import type { UserConfig as VitestConfig } from 'vitest/config';
+
+import * as vite from 'vite';
+import * as vitest from 'vitest';
+import { nodeLogDestination } from '../core/logger/node.js';
+import { openConfig } from '../core/config.js';
+import { createVite } from '../core/create-vite.js';
+import { runHookConfigDone, runHookConfigSetup } from '../integrations/index.js';
+
+export async function getViteConfig(): Promise<vite.UserConfig> {
+	let logging: LogOptions = {
+		dest: nodeLogDestination,
+		level: 'info',
+	};
+	let { astroConfig: config } = await openConfig({
+		cmd: 'test',
+		logging,
+	});
+	await runHookConfigSetup({ config, command: 'dev' });
```

Comment:
is `command: 'dev'` correct here? You use `cmd: 'test'` above so I assume that's what you want here as well?

Reply:
I switched everything to `dev` because `test` would be a new mode/command

Reply:
Is it a concern to add a new mode/command? `cmd: 'dev'` feels wrong here.

Reply:
Good point, `vitest` uses `mode: 'test'` internally so we should match that!

Reply:
Updated in [92eefee](https://github.com/withastro/astro/pull/4154/commits/92eefeedee56b940d29269601d7e916f893cebf2) to use the `mode` and `command` that Vite is resolving for us. For `vitest` usage, the **command** will be `dev` (Vite uses `serve` for clarity but we diverged) but the **mode** will be `test`.